src/main/java/org/craftercms/engine/service/context/SiteContextFactory.java

```diff
@@ -156,84 +162,44 @@ public SiteContextFactory() {
         defaultPublicBeans = new String[0];
         shutdownTimeout = DEFAULT_SHUTDOWN_TIMEOUT;
         publishingTargetMacroName = DEFAULT_PUBLISHING_TARGET_MACRO_NAME;
-    }
-
-    @Override
-    public void setServletContext(ServletContext servletContext) {
-        this.servletContext = servletContext;
-    }
 
-    public void setSiteNameMacroName(String siteNameMacroName) {
-        this.siteNameMacroName = siteNameMacroName;
-    }
-
-    @Required
-    public void setStoreType(String storeType) {
         this.storeType = storeType;
-    }
-
-    @Required
-    public void setRootFolderPath(String rootFolderPath) {
         this.rootFolderPath = rootFolderPath;
-    }
-
-    @Required
-    public void setStaticAssetsPath(String staticAssetsPath) {
         this.staticAssetsPath = staticAssetsPath;
-    }
-
-    @Required
-    public void setTemplatesPath(String templatesPath) {
         this.templatesPath = templatesPath;
-    }
-
-    @Required
-    public void setInitScriptPath(String initScriptPath) {
         this.initScriptPath = initScriptPath;
-    }
-
-    @Required
-    public void setRestScriptsPath(String restScriptsPath) {
         this.restScriptsPath = restScriptsPath;
-    }
-
-    @Required
-    public void setControllerScriptsPath(final String controllerScriptsPath) {
         this.controllerScriptsPath = controllerScriptsPath;
-    }
-
-    @Required
-    public void setConfigPaths(String[] configPaths) {
         this.configPaths = configPaths;
-    }
-
-    @Required
-    public void setApplicationContextPaths(String[] applicationContextPaths) {
         this.applicationContextPaths = applicationContextPaths;
-    }
-
-    @Required
-    public void setUrlRewriteConfPaths(String[] urlRewriteConfPaths) {
         this.urlRewriteConfPaths = urlRewriteConfPaths;
-    }
-
-    @Required
-    public void setProxyConfigPaths(String[] proxyConfigPaths) {
         this.proxyConfigPaths = proxyConfigPaths;
+        this.groovyClassesPath = groovyClassesPath;
+        this.groovyGlobalVars = groovyGlobalVars;
+        this.freeMarkerConfigFactory = freeMarkerConfigFactory;
+        this.urlTransformationEngine = urlTransformationEngine;
+        this.storeService = storeService;
+        this.cacheTemplate = cacheTemplate;
+        this.macroResolver = macroResolver;
+        this.jobResolvers = jobResolvers;
+        this.jobThreadPoolExecutor = jobThreadPoolExecutor;
+        this.graphQLFactory = graphQLFactory;
+        this.cacheWarmUpEnabled = cacheWarmUpEnabled;
+        this.cacheWarmer = cacheWarmer;
+        this.configurationReader = configurationReader;
     }
 
-    public void setTranslationConfigPaths(String[] translationConfigPaths) {
-        this.translationConfigPaths = translationConfigPaths;
+    @Override
+    public void setServletContext(ServletContext servletContext) {
```

Comment:
@sumerjabri For this particular bean, I didn't remove the setter because it is not marked with `@Required`. The migration only uses constructors for properties that are annotated with `@Required` (for simplicity and easier to track the change). We could do it better by using constructors for the case without `@Required` annotation and it is not used anywhere outside of bean initialization (and there are no circular dependencies in the bean initialization). But I hold this for another chance (could be after making sure the Spring 6 is up and running properly, or another refactoring ticket). Let me know your thoughts on this!

Reply:
I understand and agree.